AndroidSDKCore/src/main/java/com/leanplum/internal/Request.java

```diff
@@ -461,18 +457,30 @@ private void sendNow() {
       return;
     }
 
-    this.sendEventually();
+    if (!sent && !LeanplumEventDataManager.willSendErrorLog) {
+      sent = true;
+      Map<String, Object> args = createArgsDictionary();
+      saveRequestForLater(args, null);
+    }
 
     Util.executeAsyncTask(true, new AsyncTask<Void, Void, Void>() {
       @Override
       protected Void doInBackground(Void... params) {
-        sendRequests();
+        sendRequests(false);
         return null;
       }
     });
   }
 
-  private void sendRequests() {
+  /**
+   * Because saveRequestForLater() can send some requests now, we want to send only completed
+   * batches with 1000 events, not incomplete batches with few events.
+   * Due to lack of sendRequests(int limit) method, we use flag to simulate such behavior.
+   *
+   * @param onlyFirstBatch - by default is false. If true - we send only first batch of events and
+   * stop there.
+   */
+  private void sendRequests(boolean onlyFirstBatch) {
```

Comment:
I think this naming is confusing. Similar to sendNow(false), when I see sendRequests(false) I’m just confused. Is this sending request or not? You need to read the doc to understand that this flag indicates one time only request. I will make make this is a helper function and have a 2 functions that  calls it.

Reply:
Sure, feel free to suggest a better version